File: example/java/config/ai/Triggerticktype.java
package config.ai;

public enum Triggerticktype {
    CONSTVALUE("ConstValue"),
    BYLEVEL("ByLevel"),
    BYSERVERUPDAY("ByServerUpDay");

    private final String value;

    Triggerticktype(String value) {
        this.value = value;
    }

    private static final java.util.Map<String, Triggerticktype> map = new java.util.HashMap<>();

    static {
        for(Triggerticktype e : Triggerticktype.values()) {
            map.put(e.value, e);
        }
    }

    public static Triggerticktype get(String value) {
        return map.get(value);
    }

    /**
     * 枚举名称
     */
    public String getEnumName() {
        return value;
    }

}

File: example/lua/cfg/ai/triggerticktype.lua
local cfg = require "cfg._cfgs"

---@class cfg.ai.triggerticktype
---@field enumName string , 枚举名称
---@field get fun(enumName:string):cfg.ai.triggerticktype
---@field ConstValue cfg.ai.triggerticktype
---@field ByLevel cfg.ai.triggerticktype
---@field ByServerUpDay cfg.ai.triggerticktype
---@field all table<any,cfg.ai.triggerticktype>

local this = cfg.ai.triggerticktype

local mk = cfg._mk.table(this, { { 'all', 'get', 1 }, }, 1, nil, 
    'enumName' -- str, 枚举名称
    )

mk("ConstValue")
mk("ByLevel")
mk("ByServerUpDay")

return this
